src/main/java/com/graphhopper/converter/api/NominatimEntry.java

```diff
@@ -27,14 +28,39 @@ public NominatimEntry(long osmId, String type, double lat, double lon, String di
         this.displayName = displayName;
 
         this.address = new Address();
-        address.setCountry(country);
-        address.setCity(city);
+        address.country = country;
+        address.city = city;
     }
 
     public NominatimEntry() {
         this.address = new Address();
     }
 
+    public boolean isStreet() {
+        if ("highway".equals(this.classString)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns the Street name if this entry is a street, return null otherwise.
+     * We return null, since we do not serialize null properties.
+     */
+    public String getStreetNameOrNull() {
```

Comment:
I would rename to just `getStreetName` as we return `null` in other cases too without having the method named appropriate (applies to the getStreetNameOrNull method below too)